src/components/Header/Header.tsx

```diff
@@ -1,23 +1,66 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect, useRef } from 'react';
 import Link from 'next/link';
 
 import styles from './Header.module.scss';
 
 export const Header = () => {
     const [isMenuOpen, setIsMenuOpen] = useState(false);
+    const menuRef = useRef<HTMLDivElement | null>(null); // Reference for the menu
+    const closeButtonRef = useRef<HTMLButtonElement | null>(null); // Reference for the close button
+    const firstFocusableElementRef = useRef(null); // First focusable element in the menu
 
     const toggleMenu = () => {
-        setIsMenuOpen(!isMenuOpen);
+        setIsMenuOpen((prev) => !prev);
     };
 
+    // Handle trapping focus
+    useEffect(() => {
+        const handleKeyDown = (e: KeyboardEvent) => {
+            if (isMenuOpen) {
+                const focusableElements = menuRef.current?.querySelectorAll<HTMLElement>('a, button') || [];
+                const firstElement = focusableElements[0];
+                const lastElement = focusableElements[focusableElements.length - 1];
```

Comment:
Issue: It looks like the lint task is having an issue with this line being too long - we can fix this automatically by running the `npm run format` command, which is generally good to do before making a commit. You can then test to see if the lint subtask will succeed locally by running `npm run lint` and making sure no warnings appear.

Another good approach is to install the [prettier VSCode extension](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode) if you're using this development tool. You can then configure this to automatically apply formatting on file save / show red squiggles and tell you want the errors are in the code editor itself, which helps save time.